src/python/WMCore/MicroService/Unified/MSOutputTemplate.py

```diff
@@ -0,0 +1,228 @@
+        """
+
+        # check if we can fit all the arguments provided through **kwargs
+        for kw, arg in kwargs.items():
+            found = False
+            typeok = False
+            for tup in docTemplate:
+                if kw == tup[0]:
+                    found = True
+                    # NOTE: Here we can allow more than one type per field if we
+                    #       set them as a list of types eg. [str, unicode]
+                    if isinstance(kwargs[kw], tup[2]) or kwargs[kw] is None:
+                        typeok = True
+                        if update:
+                            # NOTE: Here we may consider deepcopy
+                            myDoc[kw] = kwargs[kw]
+            if not found:
+                # NOTE: We can raise an error here and decide if we want to drop
+                #       the whole document or we can jut ignore the current field
+                # NOTE: We should keep in mind that raising an error from the object
+                #       Creates a reference which will stay outside the normal
+                #       scope of the object (until not caught) which may prevent
+                #       GC cleaning the object
+                msg = "ERROR: Unrecognized parameter: {}: {}".format(kw, kwargs[kw])
+                if throw:
+                    raise KeyError(msg)
+                else:
+                    return False
+            if not typeok:
+                # NOTE: Same as above
+                msg = "ERROR: Wrong type: {} for parameter: {}: {}".format(
+                    type(kwargs[kw]),
+                    kw,
+                    kwargs[kw])
+                if throw:
+                    raise TypeError(msg)
+                else:
+                    return False
+        return True
+
+    def _checkValid(self, myDoc, throw=False):
+        """
+        An internal method to be used in order to check for some mandatory fields
+        before announcing the created document for valid
+        """
+        valid = []
+        missing = []
+        for mandField in self.required:
+            if mandField in myDoc.keys() and myDoc[mandField]:
+                valid.append(True)
+            else:
+                valid.append(False)
+                missing.append(mandField)
+
+        for mandField in self.allowEmpty:
+            if mandField in myDoc.keys():
+                valid.append(True)
+            else:
+                valid.append(False)
+                missing.append(mandField)
+
+        if not all(valid):
+            # NOTE: Same as above
+            msg = "ERROR: Missing Mandatory Fields: {} for: {}".format(missing, template)
+            if throw:
+                raise KeyError(msg)
+            else:
+                return False
+        return True
+
+    def setRelVal(self, isRelVal):
+        myDoc = {'isRelVal': isRelVal}
+        if self._checkAttr(self.docTemplate, myDoc, throw=False, update=False, **myDoc):
```

Comment:
Do I read it right that it validates the whole document before it updates the `isRelVal` attribute?

Reply:
No, you are wrong. (Sorry to say that :) ) It is actually checking only the portion that  is about to be updated. And in this case this is: `myDoc = {'isRelVal': isRelVal}` . Notice in the call:
```
if self._checkAttr(self.docTemplate, myDoc, throw=False, update=False, **myDoc):
```
It checks myDoc  against itself without updating values and without throwing an exception - just getting the `bool` value from passing everything though the `self.docTemplate`.